src/lib.rs

```diff
@@ -1311,6 +1315,112 @@ where
     Some(Ratio::new(n1, d1))
 }
 
+#[cfg(all(feature = "bigint", feature = "std"))]
+impl<T: Clone + Integer + Signed + ToPrimitive + ToBigInt> Ratio<T> {
+    /// Converts the ratio to an `f64`.
+    ///
+    /// The value is guaranteed to be the closest `f64` approximating the ratio. This always
+    /// returns a value, unless the conversion to BigInt for the numerator or denominator fails.
+    ///
+    /// ```
+    /// use num_rational::Rational64;
+    /// assert_eq!(0.5f64, Rational64::new(1, 2).to_f64().unwrap());
```

Comment:
Example uses `u32` instead of `BigInt`

Reply:
Do you think that's an issue? This PR also adds functionality for ratios of primitive integer types, and the example here seems like the simplest demonstration of what it can do.

However I can change `Rational64` to `BigRational` if you think that would be better.

Reply:
I think it's reasonable since it's an example that needs the `ToBigInt` conversion. But maybe there could *also* be a large example, with a numerator and/or denominator that can't fit in any primitive integer -- perhaps a googol (10<sup>100</sup>).

(Be careful about `assert_eq!` on floats if it's not sure to be exact though...)

Reply:
Added it. That's a good point but I think `assert_eq!` on floats in this case is fine, because this algorithm is supposed to get the exact closest float to the ratio.